tasks/discover-granules/tests/index.js

```diff
@@ -259,3 +277,148 @@ test('discover granules using S3 throws error when discovery fails',
     });
     await t.throwsAsync(() => assert(t), { code: 'NoSuchBucket' });
   });
+
+test.serial('handleDuplicates filters on duplicateHandling set to "skip"',
+  async (t) => {
+    let checkDuplicateRevert;
+    try {
+      const handleDuplicates = discoverGranulesRewire.__get__('handleDuplicates');
+      checkDuplicateRevert = discoverGranulesRewire.__set__('checkDuplicate', checkDuplicateRewire);
+      const actual = await handleDuplicates(t.context.filesByGranuleId, 'skip');
+      delete t.context.filesByGranuleId.duplicate;
+      t.deepEqual(actual, t.context.filesByGranuleId);
+    } finally {
+      checkDuplicateRevert();
+    }
+  });
+
+test.serial('handleDuplicates throws Error on duplicateHandling set to "error"',
+  async (t) => {
+    let checkDuplicateRevert;
+    try {
+      const handleDuplicates = discoverGranulesRewire.__get__('handleDuplicates');
+      checkDuplicateRevert = discoverGranulesRewire.__set__('checkDuplicate', checkDuplicateRewire);
+      await t.throwsAsync(
+        () => handleDuplicates(t.context.filesByGranuleId, 'error')
+      );
+    } finally {
+      checkDuplicateRevert();
+    }
+  });
+
+test('handleDuplicates throws Error on an invalid duplicateHandling configuration',
+  async (t) => {
+    const handleDuplicates = discoverGranulesRewire.__get__('handleDuplicates');
+    await t.throwsAsync(
+      () => handleDuplicates(t.context.filesByGranuleId, 'foobar')
+    );
+  });
+
+test.serial('handleDuplicates does not filter when duplicateHandling is set to "replace" or "version"',
+  async (t) => {
+    let checkDuplicateRevert;
+    try {
+      const handleDuplicates = discoverGranulesRewire.__get__('handleDuplicates');
+      checkDuplicateRevert = discoverGranulesRewire.__set__('checkDuplicate', checkDuplicateRewire);
+      const replaceActual = await handleDuplicates(t.context.filesByGranuleId, 'replace');
+      const versionActual = await handleDuplicates(t.context.filesByGranuleId, 'version');
+      t.deepEqual(replaceActual, t.context.filesByGranuleId);
+      t.deepEqual(versionActual, t.context.filesByGranuleId);
+    } finally {
+      checkDuplicateRevert();
+    }
+  });
+
+
+test.serial('filterDuplicates returns a set of filtered keys',
+  async (t) => {
+    let checkDuplicateRevert;
+    try {
+      const filterDuplicates = discoverGranulesRewire.__get__('filterDuplicates');
+      checkDuplicateRevert = discoverGranulesRewire.__set__('checkDuplicate', async (key) => {
+        if (key === 'duplicate') {
+          return '';
```

Comment:
So is this relying on `''` being treated as falsy by `.filter(Boolean)`?

Reply:
Correct - we could return false here, but that doesn't map to the function's behavior.

Reply:
....except.... that's not true.  Cool.

Reply:
93e70f3